proposals/nnnn-rpc-design-refactoring.md

```diff
@@ -0,0 +1,153 @@
+# RPC design refactoring
+
+* Proposal: [SDL-NNNN](nnnn-rpc-design-refactoring.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+This proposal is to rework design of application manager layer in order to:
+ - Reduce coupling between components
+ - Reduce responsibility of ApplicationManager class
+ - Provide plugin architecture for easy adding additional functionality and RPC groups   
+
+## Motivation
+
+SDL core project is actively implementing new features and new functionality.  
+To keep project on track and to reduce risks of regression during implementation
+refactoring of application manager layer should be done.  
+
+Currently, extending SDL functionality may require code duplication. 
+RC module, in fact, duplicates logic of application_manager layer, there is separate request controller, separate mechanism to call  polcies checks, and RPC's processing. So any common logic that should be implemented in application_manager layer should be duplicated in request controller, it results in bugs, and requires more time for implementation. 
```

Comment:
So any common logic that should be implemented in application_manager layer should be duplicated in request controller, it results in bugs, and requires more time for implementation.
should become
Currently any common logic that is implemented in the application_manager layer needs to be duplicated in request controller -- this results in bugs and redundant code; it also requires more time for implementation.

Reply:
@mrapitis, done in 2c9f7a3f7bb7e6c8b26bb1e4aec779cf65733666

Reply:
Can we double check these changes also? Its important that we state the original RC components will be refactored first.